sdk/core/aborter/src/aborter.ts

```diff
@@ -0,0 +1,284 @@
+/// <reference lib="es5" />
+/// <reference lib="dom" />
+
+import { AbortSignalLike, isNode } from "@azure/ms-rest-js";
+
+/**
+ * An aborter instance implements AbortSignal interface, can abort HTTP requests.
+ *
+ * - Call Aborter.none to create a new Aborter instance without timeout.
+ * - Call Aborter.timeout() to create a new Aborter instance with timeout.
+ *
+ * For an existing instance aborter:
+ * - Call aborter.withTimeout() to create and return a child Aborter instance with timeout.
+ * - Call aborter.withValue(key, value) to create and return a child Aborter instance with key/value pair.
+ * - Call aborter.abort() to abort current instance and all children instances.
+ * - Call aborter.getValue(key) to search and get value with corresponding key from current aborter to all parents.
+ *
+ * @example
+ * // Abort without timeout
+ * await blockBlobClient.upload(Aborter.none, buf, buf.length);
```

Comment:
We might need to update examples as `blockBlobClient` is storage specific.

Reply:
That's fair, I think I should be able to adapt the examples in the readme to work for this documentation as well.

Reply:
Updated to be similar to the examples in the README.